docs/permission/writing-a-policy.md

```diff
@@ -84,59 +98,68 @@ In addition to the conditions provided by the catalog plugin, you can write your
 Now let's say we would also like to restrict users from viewing catalog entities that they do not own, just like we did for unregistering entities. One way to achieve this may be to simply duplicate our if statement and check for the `catalog.entity.read` permission:
 
 ```diff
-    async handle(
-      request: PolicyAuthorizeQuery,
-      user?: BackstageIdentityResponse,
-     ): Promise<PolicyDecision> {
-      if (request.permission.name === 'catalog.entity.delete') {
-        return createCatalogPolicyDecision(
+  async handle(
+    request: PolicyQuery,
+    user?: BackstageIdentityResponse,
+  ): Promise<PolicyDecision> {
+    if (isResourcePermission(request.permission, 'catalog-entity')) {
+      if (request.permission.name === 'catalog.entity-delete') {
```

Comment:
instead of checking the resource permission and then the names we could simplify to `isPermission(request.permission, catalogEntityDeletePermission)`, which will also narrow `request.permission` to `ResourceType<'catalog-entity'>`